lhotse/recipes/mgb2.py

```diff
@@ -0,0 +1,202 @@
+                 | sed 's:wav/:{path.join(corpus_dir, part,'wav/')}:g' > {path.join(output_dir, part,'wav.scp')}") 
+
+            recordings, supervisions, _= load_kaldi_data_dir(path.join(output_dir, part), 16000)
+            if buck_walter == False:
+                supervisions = supervisions.transform_text(fromBuckWalter)
+            if part == "test":
+                assert len(supervisions) == 5365
+            elif part == "dev":
+                assert len(supervisions) == 5002
+        elif part == "train":
+            recordings = RecordingSet.from_dir(path.join(corpus_dir, part, "wav"),
+                                                pattern='*.wav', num_jobs=num_jobs)
+
+            xml_paths = check_and_rglob(path.join(corpus_dir, part, "xml/utf8"), "*.xml")
+            # Read supervisions and write them to manifest
+            with recursion_limit(5000):
+                supervisions_list = list(chain.from_iterable(
+                    [make_supervisions(p, mer_thresh) for p in xml_paths]))
+            
+            supervisions = SupervisionSet.from_segments(supervisions_list)
+            
+            assert len(supervisions) == 375103
+                
+            if text_cleaning == True:
+                supervisions = supervisions.transform_text(cleaning)
+            recordings, supervisions = fix_manifests(recordings, supervisions)
+        validate_recordings_and_supervisions(recordings, supervisions)
+
+        # saving recordings and supervisions
+        recordings.to_file(path.join(output_dir,f"mgb2_recordings_{part}.jsonl.gz"))
+        supervisions.to_file(path.join(output_dir,f"mgb2_supervisions_{part}.jsonl.gz"))
+
+        manifests[part] = {
+                "recordings":recordings,
+                "supervisions": supervisions,
+                }
+    return manifests
+
+_unicode = u"\u0622\u0624\u0626\u0628\u062a\u062c\u06af\u062e\u0630\u0632\u0634\
+    \u0636\u0638\u063a\u0640\u0642\u0644\u0646\u0648\u064a\u064c\u064e\u0650\u0652\u0670\
+    \u067e\u0686\u0621\u0623\u0625\u06a4\u0627\u0629\u062b\u062d\u062f\u0631\u0633\u0635\
+    \u0637\u0639\u0641\u0643\u0645\u0647\u0649\u064b\u064d\u064f\u0651\u0671"
+_buckwalter = u"|&}btjGx*z$DZg_qlnwyNaio`PJ'><VApvHdrsSTEfkmhYFKu~{"
+_backwardMap = {ord(a):b for a,b in zip(_buckwalter, _unicode)}
+
+def fromBuckWalter(s: str) -> str:
+    return s.translate(_backwardMap)
+
+def remove_diacritics(text: str) -> str:
+    #https://unicode-table.com/en/blocks/arabic/
+    return sub(r'[\u064B-\u0652\u06D4\u0670\u0674\u06D5-\u06ED]+', '', text)
+
+def remove_punctuations(text: str) -> str:
+	""" This function  removes all punctuations except the verbatim """
+	
+	arabic_punctuations = '''`÷×؛<>_()*&^%][ـ،/:"؟.,'{}~¦+|!”…“–ـ'''
+	english_punctuations = punctuation
+	all_punctuations = set(arabic_punctuations + english_punctuations) # remove all non verbatim punctuations
+	
+	for p in all_punctuations:
+		if p in text:
```

Comment:
Should it be indented with 4 spaces, not 8?

Reply:
Done

Reply:
Done